cli/src/main/java/org/projectnessie/tools/catalog/migration/cli/CatalogMigrationCLI.java

```diff
@@ -0,0 +1,507 @@
+    if (sourceHadoopConf != null && !sourceHadoopConf.isEmpty()) {
+      sourceHadoopConf.forEach(sourceCatalogConf::set);
+    }
+    Catalog sourceCatalog =
+        CatalogUtil.loadCatalog(
+            Objects.requireNonNull(catalogImpl(sourceCatalogType, sourceCustomCatalogImpl)),
+            sourceCatalogType.name(),
+            sourceCatalogProperties,
+            sourceCatalogConf);
+    printWriter.println(String.format("%nConfigured source catalog: %s", sourceCatalogType.name()));
+
+    Configuration targetCatalogConf = new Configuration();
+    if (targetHadoopConf != null && !targetHadoopConf.isEmpty()) {
+      targetHadoopConf.forEach(targetCatalogConf::set);
+    }
+    Catalog targetCatalog =
+        CatalogUtil.loadCatalog(
+            Objects.requireNonNull(catalogImpl(targetCatalogType, targetCustomCatalogImpl)),
+            targetCatalogType.name(),
+            targetCatalogProperties,
+            targetCatalogConf);
+    printWriter.println(String.format("%nConfigured target catalog: %s", targetCatalogType.name()));
+
+    List<TableIdentifier> tableIdentifiers = processIdentifiersInput(printWriter);
+
+    if (!canProceed(printWriter, sourceCatalog)) {
+      return 0;
+    }
+
+    ImmutableCatalogMigratorParams params =
+        ImmutableCatalogMigratorParams.builder()
+            .sourceCatalog(sourceCatalog)
+            .targetCatalog(targetCatalog)
+            .deleteEntriesFromSourceCatalog(deleteSourceCatalogTables)
+            .build();
+    CatalogMigrator catalogMigrator = new CatalogMigrator(params);
+
+    List<TableIdentifier> identifiers;
+    if (tableIdentifiers.isEmpty()) {
+      if (identifiersRegEx == null) {
+        printWriter.println(
+            String.format(
+                "%nUser has not specified the table identifiers."
+                    + " Selecting all the tables from all the namespaces from the source catalog."));
+      } else {
+        printWriter.println(
+            String.format(
+                "%nUser has not specified the table identifiers."
+                    + " Selecting all the tables from all the namespaces from the source catalog "
+                    + "which matches the regex pattern:"
+                    + identifiersRegEx));
+      }
+      identifiers = catalogMigrator.getMatchingTableIdentifiers(identifiersRegEx);
+    } else {
+      identifiers = tableIdentifiers;
+    }
+
+    String operation = deleteSourceCatalogTables ? "migration" : "registration";
+    printWriter.println(
+        String.format("%nIdentified %d tables for %s.", identifiers.size(), operation));
+
+    ImmutableCatalogMigrationResult.Builder resultBuilder =
+        ImmutableCatalogMigrationResult.builder();
+    if (isDryRun) {
+      CatalogMigrationResult result =
+          resultBuilder.addAllRegisteredTableIdentifiers(identifiers).build();
+      writeToFile(pathWithOutputDir(DRY_RUN_FILE), result.registeredTableIdentifiers());
```

Comment:
Adding `identifiers` to that builder for the sake of pulling it out in the next line is overly complicated.